packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -27,33 +27,51 @@ import {
 } from "../../core";
 import { DefaultChangeset, DefaultEditBuilder } from "../defaultChangeFamily";
 import { runSynchronousTransaction } from "../defaultTransaction";
+import { singleMapTreeCursor } from "../mapTreeCursor";
 import { ProxyTarget, EditableField, proxifyField, UnwrappedEditableField } from "./editableTree";
+import { applyFieldTypesFromContext, ContextuallyTypedNodeData } from "./utilities";
 
 /**
  * A common context of a "forest" of EditableTrees.
  * It handles group operations like transforming cursors into anchors for edits.
  */
 export interface EditableTreeContext {
     /**
-     * Gets a Javascript Proxy providing a JavaScript object like API for interacting with the tree.
+     * Gets or sets the root field of the tree.
      *
-     * Use built-in JS functions to get more information about the data stored e.g.
-     * ```
-     * for (const key of Object.keys(context.root)) { ... }
-     * // OR
-     * if ("foo" in data) { ... }
-     * context.free();
-     * ```
-     *
-     * Not (yet) supported: create properties, set values and delete properties.
+     * Its setter works exactly the same way as {@link EditableTreeContext.unwrappedRoot}.
      */
-    readonly root: EditableField;
+    get root(): EditableField;
+
+    set root(data: ContextuallyTypedNodeData | undefined);
 
     /**
-     * Same as `root`, but with unwrapped fields.
-     * See ${@link UnwrappedEditableField} for what is unwrapped.
+     * Gets or sets the root field of the tree.
+     *
+     * When using its getter, see {@link UnwrappedEditableField} for what is unwrapped.
+     *
+     * When using its setter, the input data must be formed depending
+     * on a multiplicity of the field, on if it's polymorphic or not and, for non-sequence multiplicities,
+     * on if the field's node declares its primary function using a primary field (see `getPrimaryField`):
+     * - For `Sequence` multiplicities and "primary fielded" nodes, array data or an {@link EditableField} is expected.
+     * Use empty array (`[]`) to delete all nodes of a sequence field.
+     * - For `Optional` multiplicities, `ContextuallyTypedNodeDataObject | undefined` is expected.
+     * If the input data is `undefined`, the field node will be deleted if it exists.
+     * - For `Value` multiplicities, `ContextuallyTypedNodeDataObject` is expected.
+     *
+     * If the field is a non-sequence and some of its types declare to follow
+     * the `String`, `Number` or `Boolean` value schema (see `ValueSchema`),
+     * a `PrimitiveValue` can be used instead of a `ContextuallyTypedNodeDataObject`
+     * to create/replace or to set the value of the primitive node.
+     * Required is to be possible to unambiguously resolve the node type out of a primitive type
```

Comment:
```suggestion
     * For this to work, it must be possible to unambiguously resolve the node type to a primitive type
```

Reply:
I have reworked this to be (hopefully) easier to understand